pkg/status/statuscollector.go

```diff
@@ -50,3 +65,92 @@ func (c *Controller) syncStatusCollector(ctx context.Context, ref string) error
 	logger.Info("Synced StatusCollector", "ref", ref)
 	return nil
 }
+
+func (c *Controller) validateStatusCollector(statusCollector *v1alpha1.StatusCollector) []error {
+	var errs []error
+	// groupBy & CombinedFields empty if select is not
+	if len(statusCollector.Spec.Select) > 0 &&
+		(len(statusCollector.Spec.GroupBy) > 0 || len(statusCollector.Spec.CombinedFields) > 0) {
+		errs = append(errs, fmt.Errorf("groupBy and combinedFields must be empty if select is not"))
+	}
+	// groupBy empty if combinedFields is
+	if len(statusCollector.Spec.CombinedFields) == 0 && len(statusCollector.Spec.GroupBy) > 0 {
+		errs = append(errs, fmt.Errorf("groupBy must be empty if combinedFields is"))
+	}
+
+	// structure must be valid before we get to parsing errors
+	if len(errs) > 0 {
+		return errs
+	}
+
+	// validate filter expression
+	if err := c.celEvaluator.CheckExpression(statusCollector.Spec.Filter); err != nil {
+		errs = append(errs, fmt.Errorf("filter expression invalid: %w", err))
+	}
+
+	// validate select expression
+	for _, selectExpr := range statusCollector.Spec.Select {
+		if err := c.celEvaluator.CheckExpression(&selectExpr.Def); err != nil {
+			errs = append(errs, fmt.Errorf("select expression (%s) invalid: %w", selectExpr.Name, err))
+		}
+	}
+
+	// validate groupBy expression
+	for _, groupByExpr := range statusCollector.Spec.GroupBy {
+		if err := c.celEvaluator.CheckExpression(&groupByExpr.Def); err != nil {
+			errs = append(errs, fmt.Errorf("groupBy expression (%s) invalid: %w", groupByExpr.Name, err))
+		}
+	}
+
+	// validate combinedFields expression
+	for _, combinedField := range statusCollector.Spec.CombinedFields {
+		if combinedField.Type == v1alpha1.AggregatorTypeCount {
+			if combinedField.Subject != nil {
+				errs = append(errs,
+					fmt.Errorf("combinedField expression (%s) invalid: subject must be nil for %s type",
+						combinedField.Name, combinedField.Type))
+			}
+
+			continue
+		}
+
+		if combinedField.Subject == nil {
+			errs = append(errs, fmt.Errorf("combinedField expression (%s) invalid: subject must be set",
+				combinedField.Name))
+			continue
+		}
+
+		if err := c.celEvaluator.CheckExpression(combinedField.Subject); err != nil {
+			errs = append(errs, fmt.Errorf("combinedField expression (%s) subject invalid: %w",
+				combinedField.Name, err))
+		}
+	}
+
+	return errs
+}
+
+func (c *Controller) updateStatusCollectorErrors(ctx context.Context, statusCollector *v1alpha1.StatusCollector,
```

Comment:
Since the value of `statusCollector` here comes from https://github.com/kubestellar/kubestellar/blob/08d290a0dbcfe91714d7e140cd80630c5ed843b9/pkg/generated/listers/control/v1alpha1/statuscollector.go#L36 , it is worth noting its immutability.